### PR TITLE
[Assignment 3] 100% Coverage

### PR DESCRIPTION
**Pull request by sfabriece**

### CODE REVIEW

File: lib/start.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var Index = require('./index');
+var Hoek = require('hoek');
+Index.init(8000, function(err, server){
+    Hoek.assert(!err, err);
+    console.log('Server started at: ' + server.info.uri);
+    return server;
```
Comment (chyld) [inline lib/start.js]: don't return the server

**Comment by sfabriece** [comment]
done :)

**Comment by EvanMarkHopkins** [comment]
I believe the -L flag in lab enforces these styles by default, so you don't need to specify them yourself. Also, the 4th objective specified a makefile, there's nothing wrong with what you have done, but it's worth giving a shot.

**Comment by sfabriece** [comment]
added makefile, can you elaborate on why it is preferable to use a makefile?

File: lib/start.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+var Index = require('./index');
+var Hoek = require('hoek');
+Index.init(8000, function(err, server){
```
Comment (AdriVanHoudt) [inline lib/start.js]: newline after function and also between modules and the init() function call

File: package.json [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
-  "name": "hueniversity",
-  "version": "0.0.2",
-  "description": "Community learning experiment",
-  "main": "lib/index.js",
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/hueniverse/hueniversity.git"
-  },
-  "keywords": [
+	"name": "hueniversity",
```
Comment (AdriVanHoudt) [inline package.json]: why did you change the indentation?

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+var Code = require('code');
+var Lab = require('lab');
+var Server = require('../lib');
+var Version = require('../lib/version');
+
+var expect = Code.expect;
+var labi = exports.lab = Lab.script();
+
+labi.experiment('index', function(){
+    labi.test('init starts the server', function(done){
```
Comment (AdriVanHoudt) [inline test/index.js]: newline after function

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+        Server.init(8081, function(err, server){
```
Comment (AdriVanHoudt) [inline test/index.js]: better to not pass along a defined port since that might interfere with other tests

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    labi.test('should report error if plugin registration fails', function(done){
+
+        var correctVersionPlugin = Version.register;
+        var fakePluginError = 'Fake Plugin Error';
+        Version.register = function(server, options, next){
+
+            return next(fakePluginError);
+        };
+        Version.register.attributes = {name: 'Fake version Plugin' };
```
Comment (AdriVanHoudt) [inline test/index.js]: space after {

File: test/version.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+        Server.init(8000, function(err, server){
+
+            var options = {
+                url: '/version'
+            };
+            server.inject(options, function(res){
+
+                expect(res.result).to.be.an.object();
+                expect(res.result.version).to.equal(Pkg.version);
```
Comment (AdriVanHoudt) [inline test/version.js]: what if the server responds with an object with an additional (not wanted) key? The test would still pass...

**Comment by AdriVanHoudt** [comment]
@sfabriece because learning and Eran likes it ^^ also @EvanMarkHopkins is right when using -L lab will use predefined rules for Hapi that come along with it

File: package.json (hunk 3 above)
Comment (EvanMarkHopkins) [inline package.json]: maybe an editor problem? it seems to handle JS pretty well, but perhaps some special attention to JSON is in order. Requiring a final newline in files is another thing to look into.

Comment (sfabriece) [inline package.json]: an editor problem and I forgot to run eslint.

**Comment by zoe-1** [comment]
@sfabriece The Makefile is useful for automating tasks in your project. 
npm commands can execute automated tasks but they can get very long.  Long commands are inconvenient to type over and over.  

For example, npm allows you to run two commands related to executing scripts by default: "npm start" and "npm test".  If you make other custom npm commands in the scripts section, they will fail without "run" preceding the command.
ex) "npm run command_name" // illustrates a command with run before it.

See below command result for npm's default commands that do not need a "run" before them.
npm --help command  // This will list npm's commands (test and start are included here).

Look at: https://github.com/zoe-1/nmax/blob/master/package.json to see a package.json file with a significant amount of package.json scripts commands in them. 
The commands are made in the scripts:{ } section.
In the above linked package.json file there is the below command configured:
"protractor": "protractor test/protractor-conf.js"
To execute this command you must type: "npm run protractor". "npm protractor" without "run" will not work . "npm run protractor" will run tests on the project. The command is a long command to type every time you want to run tests.  The Makefile lets you simplify this.  
Put in your Makefile:
----<br/>
`pro:`
&nbsp;&nbsp;&nbsp;&nbsp;`npm run protractor`<br/>


Then type:<br/>
&nbsp;&nbsp;&nbsp;&nbsp;`make pro`<br/>

And, the long "npm run protractor" command will be executed.  

As a project develops you will need automate tasks like: 
1. changes to the configuration state (live versus development states)
2. execution of tests
3. server start up 
4. and others....(For example, git push commands to different remote repos) 

In this assignment, I believe Eran is showing us how to use the Makefile as a convenient method
to manage and execute our projects most important automated tasks.

**Comment by sfabriece** [comment]
@zoe-1 Thanks for the explanation.

File: test/index.js (hunk 6 above)
Comment (sfabriece) [inline test/index.js]: Should this be caught by the linter?

File: test/index.js (hunk 5 above)
Comment (sfabriece) [inline test/index.js]: does deep equality test solve this?

Comment (AdriVanHoudt) [inline test/index.js]: what do you mean? The problem is that when you write other tests and you use the same port there by accident it will cause issues, so better to not use specified ports unless for the test whether init() will use the specified port

File: test/index.js (hunk 6 above)
Comment (AdriVanHoudt) [inline test/index.js]: yes but the lint rules are not complete(feel free to add to them https://github.com/hapijs/lab/blob/master/lib/linters/eslint/.eslintrc )
